<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Module 4 | Epigenomic Analysis 2021</title>
  <meta name="description" content="Workshop website for the 2021 Epigenomic Analysis Canadian Bioinformatics Workshop" />
  <meta name="generator" content="bookdown 0.43 and GitBook 2.6.7" />

  <meta property="og:title" content="Module 4 | Epigenomic Analysis 2021" />
  <meta property="og:type" content="book" />
  <meta property="og:image" content="https://bioinformaticsdotca.github.io/EPI_2021/img/bioinformatics_logo.png" />
  <meta property="og:description" content="Workshop website for the 2021 Epigenomic Analysis Canadian Bioinformatics Workshop" />
  <meta name="github-repo" content="bioinformaticsdotca/EPI_2021" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Module 4 | Epigenomic Analysis 2021" />
  
  <meta name="twitter:description" content="Workshop website for the 2021 Epigenomic Analysis Canadian Bioinformatics Workshop" />
  <meta name="twitter:image" content="https://bioinformaticsdotca.github.io/EPI_2021/img/bioinformatics_logo.png" />

<meta name="author" content="Faculty: Guillaume Bourque, Martin Hirst, David Bujold, Jose Hector Galvez, Edmund Su, BF Francis Ouellette, Rachade Hmamouchi" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="img/favicon.ico" type="image/x-icon" />
<link rel="prev" href="module-3.html"/>
<link rel="next" href="module-5-downstream-analyses-integrative-tools.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/6.5.1/css/all.min.css">
<script type="application/ld+json">
{
  "@context": "https://schema.org",
  "@type": "LearningResource",
  "name": "Epigenomics Analysis 2021",
  "description": "The complete set of tutorials, slides, and practical exercises for the Epigenomics Analysis 2021 Canadian Bioinformatics Workshop, hosted on GitHub Pages.",
  "url": "https://bioinformaticsdotca.github.io/EPI_2021",
  "learningResourceType": ["tutorial", "slides", "exercise"],
  "inLanguage": "en",
  "license": "https://creativecommons.org/licenses/by/4.0/",
  "provider": {
    "@type": "Organization",
    "name": "Bioinformatics.ca",
    "url": "https://bioinformatics.ca"
  },
  "keywords": "",
  "isPartOf": {
    "@type": "Course",
    "@id": "http://bioinformaticsdotca.github.io/EPI_2021"
  }
}
</script>



<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<a href="https://bioinformaticsdotca.github.io/">
 <img src="img/bioinformatics_logo.png" width="90%" alt="bioinformatics.ca logo" style="display:block; margin-left:auto; margin-right:auto; margin-top:15px;">
 </a>
<li><center><strong><a href="./"><br>Epigenomic Analysis<br><br></a></strong></center></li>
<a href="./">
 <img src="img/CBW_Epigenome-data_icon.png" width="70%" alt="Epigenomic Analysis logo" style="display:block; margin-left:auto; margin-right:auto; ">
 </a>
<br>

<li class="divider"></li>
<li class="part"><span><b>I Introduction</b></span></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Workshop Info</a>
<ul>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#pre-work"><i class="fa fa-check"></i>Pre-work</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#schedule"><i class="fa fa-check"></i>Schedule</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="meet-your-faculty.html"><a href="meet-your-faculty.html"><i class="fa fa-check"></i>Meet Your Faculty</a></li>
<li class="chapter" data-level="" data-path="compute-setup.html"><a href="compute-setup.html"><i class="fa fa-check"></i>Compute Setup</a></li>
<li class="part"><span><b>II Modules</b></span></li>
<li class="chapter" data-level="" data-path="module-1-introduction-to-chip-sequencing-and-analysis.html"><a href="module-1-introduction-to-chip-sequencing-and-analysis.html"><i class="fa fa-check"></i>Module 1: Introduction to ChIP Sequencing and Analysis</a>
<ul>
<li class="chapter" data-level="" data-path="module-1-introduction-to-chip-sequencing-and-analysis.html"><a href="module-1-introduction-to-chip-sequencing-and-analysis.html#lecture"><i class="fa fa-check"></i>Lecture</a></li>
<li class="chapter" data-level="" data-path="module-1-introduction-to-chip-sequencing-and-analysis.html"><a href="module-1-introduction-to-chip-sequencing-and-analysis.html#lab"><i class="fa fa-check"></i>Lab</a>
<ul>
<li class="chapter" data-level="" data-path="module-1-introduction-to-chip-sequencing-and-analysis.html"><a href="module-1-introduction-to-chip-sequencing-and-analysis.html#common-tools-of-the-trade"><i class="fa fa-check"></i>Common Tools of the Trade</a></li>
<li class="chapter" data-level="" data-path="module-1-introduction-to-chip-sequencing-and-analysis.html"><a href="module-1-introduction-to-chip-sequencing-and-analysis.html#module-1.-bwa-alignment-bam-post-processing"><i class="fa fa-check"></i>Module 1. BWA Alignment + BAM Post-processing</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="module-2.html"><a href="module-2.html"><i class="fa fa-check"></i>Module 2</a>
<ul>
<li class="chapter" data-level="" data-path="module-2.html"><a href="module-2.html#lecture-1"><i class="fa fa-check"></i>Lecture</a></li>
<li class="chapter" data-level="" data-path="module-2.html"><a href="module-2.html#lab-1"><i class="fa fa-check"></i>Lab</a>
<ul>
<li class="chapter" data-level="" data-path="module-2.html"><a href="module-2.html#module-2.-peak-calling"><i class="fa fa-check"></i>Module 2. Peak Calling</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="module-3.html"><a href="module-3.html"><i class="fa fa-check"></i>Module 3</a>
<ul>
<li class="chapter" data-level="" data-path="module-3.html"><a href="module-3.html#lecture-2"><i class="fa fa-check"></i>Lecture</a></li>
<li class="chapter" data-level="" data-path="module-3.html"><a href="module-3.html#lab-2"><i class="fa fa-check"></i>Lab</a>
<ul>
<li class="chapter" data-level="" data-path="module-3.html"><a href="module-3.html#visualization"><i class="fa fa-check"></i>3. Visualization</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="module-4.html"><a href="module-4.html"><i class="fa fa-check"></i>Module 4</a>
<ul>
<li class="chapter" data-level="" data-path="module-4.html"><a href="module-4.html#lecture-3"><i class="fa fa-check"></i>Lecture</a></li>
<li class="chapter" data-level="" data-path="module-4.html"><a href="module-4.html#lab-3"><i class="fa fa-check"></i>Lab</a>
<ul>
<li class="chapter" data-level="" data-path="module-4.html"><a href="module-4.html#introduction"><i class="fa fa-check"></i>1. Introduction</a></li>
<li class="chapter" data-level="" data-path="module-4.html"><a href="module-4.html#mapping-tutorial"><i class="fa fa-check"></i>2. Mapping Tutorial</a></li>
<li class="chapter" data-level="" data-path="module-4.html"><a href="module-4.html#differential-methylation-analysis-in-methylkit"><i class="fa fa-check"></i>3. Differential Methylation Analysis in MethylKit</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="module-5-downstream-analyses-integrative-tools.html"><a href="module-5-downstream-analyses-integrative-tools.html"><i class="fa fa-check"></i>Module 5: Downstream Analyses &amp; Integrative Tools</a>
<ul>
<li class="chapter" data-level="" data-path="module-5-downstream-analyses-integrative-tools.html"><a href="module-5-downstream-analyses-integrative-tools.html#lecture-4"><i class="fa fa-check"></i>Lecture</a></li>
<li class="chapter" data-level="" data-path="module-5-downstream-analyses-integrative-tools.html"><a href="module-5-downstream-analyses-integrative-tools.html#lab-4"><i class="fa fa-check"></i>Lab</a>
<ul>
<li class="chapter" data-level="" data-path="module-5-downstream-analyses-integrative-tools.html"><a href="module-5-downstream-analyses-integrative-tools.html#introduction-1"><i class="fa fa-check"></i>Introduction</a></li>
<li class="chapter" data-level="" data-path="module-5-downstream-analyses-integrative-tools.html"><a href="module-5-downstream-analyses-integrative-tools.html#lab-part-1"><i class="fa fa-check"></i>Lab Part 1</a></li>
<li class="chapter" data-level="" data-path="module-5-downstream-analyses-integrative-tools.html"><a href="module-5-downstream-analyses-integrative-tools.html#ihec-data-portal"><i class="fa fa-check"></i>IHEC Data Portal</a></li>
<li class="chapter" data-level="" data-path="module-5-downstream-analyses-integrative-tools.html"><a href="module-5-downstream-analyses-integrative-tools.html#predicting-motifs-with-homer"><i class="fa fa-check"></i>Predicting Motifs with HOMER</a></li>
<li class="chapter" data-level="" data-path="module-5-downstream-analyses-integrative-tools.html"><a href="module-5-downstream-analyses-integrative-tools.html#looking-for-go-terms-enrichment-with-great"><i class="fa fa-check"></i>Looking for GO Terms Enrichment with GREAT</a></li>
<li class="chapter" data-level="" data-path="module-5-downstream-analyses-integrative-tools.html"><a href="module-5-downstream-analyses-integrative-tools.html#going-back-to-homer-results"><i class="fa fa-check"></i>Going back to HOMER Results</a></li>
</ul></li>
</ul></li>
<li class="divider"></li>
<h1 id="sponsors">Sponsors</h1>
<center>
<a href="https://www.cihr-irsc.gc.ca"><img src="img/sponsors/cihr.png" width=90% style="padding-bottom: 20px" alt="Canadian Institutes of Health Research logo."></a>

<a href="https://genomecanada.ca/"><img src="img/sponsors/genomecanada.png" height="55" style="display:inline-block;"  style="padding-bottom: 20px" alt="Genome Canada logo.">
<a href="https://www.ontariogenomics.ca/"><img src="img/sponsors/ontariogenomics.png" height="55" style="display:inline-block;" style="padding-bottom: 20px;" alt="Ontario Genomics logo.">
<br><br>
<a href="https://oicr.on.ca/"><img src="img/sponsors/oicr.png" width=35% style="display:inline-block;" style="padding-bottom: 20px" alt="Ontario Institute for Cancer Research logo.">

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Epigenomic Analysis 2021</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="module-4" class="section level1 hasAnchor">
<h1>Module 4<a href="module-4.html#module-4" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div id="lecture-3" class="section level2 hasAnchor">
<h2>Lecture<a href="module-4.html#lecture-3" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<iframe src="https://drive.google.com/file/d/1djr2nWLHzgUwtz6o61O60bnilAUAwSy7/preview" width="640" height="480" allow="autoplay">
</iframe>
</div>
<div id="lab-3" class="section level2 hasAnchor">
<h2>Lab<a href="module-4.html#lab-3" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="introduction" class="section level3 hasAnchor">
<h3>1. Introduction<a href="module-4.html#introduction" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Description of the lab:</p>
<p>This module will cover the basics of Whole Genome Bisulfite-Sequencing (WGBS) data analysis including data visualization in IGV.</p>
<p>Objectives:</p>
<ol style="list-style-type: decimal">
<li>Learn how to align WGBS data using <code>Bismark</code> 2) Learn how to generate a methylation profile with <code>Bismark</code> 3) Learn how to open alignments and methylation profiles in the <code>IGV</code> genome browser 4) Learn how to perform a basic differential methylation analysis with <code>MethylKit</code></li>
</ol>
<p>Local software that we will use:
Before you begin, make sure you have the following programs ready in your local computer:</p>
<ul>
<li>A connection to the <code>EPI_2021</code> AWS instance</li>
<li>An internet browser</li>
<li>IGV</li>
<li>R (or RStudio), <em>optional</em></li>
</ul>
</div>
<div id="mapping-tutorial" class="section level3 hasAnchor">
<h3>2. Mapping Tutorial<a href="module-4.html#mapping-tutorial" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="getting-started" class="section level4 hasAnchor">
<h4>2.1 Getting Started<a href="module-4.html#getting-started" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Prepare Directory for the Lab</p>
<pre><code>mkdir -p ~/workspace/module4
cd ~/workspace/module4</code></pre>
<p>Save Genome Location</p>
<pre><code>GENOME=~/CourseData/EPI_data/module4/Homo_sapiens.GRCh38.chr19
export GENOME</code></pre>
<p>This will define a variable <code>$GENOME</code> that will simplify future commands.</p>
</div>
<div id="locate-the-data-for-the-workshop" class="section level4 hasAnchor">
<h4>Locate the Data for the Workshop<a href="module-4.html#locate-the-data-for-the-workshop" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<pre><code>WGBS_DATA=~/CourseData/EPI_data/module4/data
export WGBS_DATA</code></pre>
<p>This will define a variable <code>$WGBS_DATA</code> that will simplify future commands.</p>
<p>Check the Files
<strong>Question 1</strong></p>
<p><em>Type the following command:</em> <code>ls $WGBS_DATA</code>, what do you see?</p>
<p><strong>Answer</strong></p>
<p>You should see something similar to this:</p>
<pre><code>WGBS.A34002.137160.chr19.1.fastq.gz
WGBS.A34002.137160.chr19.2.fastq.gz
WGBS.A34002.137487.chr19.1.fastq.gz
WGBS.A34002.137487.chr19.2.fastq.gz
WGBS.A34002.137488.chr19.1.fastq.gz
WGBS.A34002.137488.chr19.2.fastq.gz</code></pre>
<p>These are the files that will be used for the workshop. They contain a subset of WGBS reads from CEMT sample <code>CEMT0007</code>, which is a mammary gland epithelial cell line (<a href="https://ega-archive.org/datasets/EGAD00001000685">more information here</a>).</p>
<p><strong>Question 2</strong></p>
<p><em>What do the “.1” and “.2” in the file names mean?</em></p>
<p><strong>Answer</strong></p>
<p>They represent the <code>read1</code> and <code>read2</code> of the paired end reads.</p>
</div>
<div id="map-data-using-bismark" class="section level4 hasAnchor">
<h4>2.2 Map Data using Bismark<a href="module-4.html#map-data-using-bismark" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>We will now process and map the reads using the Bismark WGBS aligner (<a href="https://www.bioinformatics.babraham.ac.uk/projects/bismark/">more info here</a>).</p>
<p>Map the first dataset using Bismark</p>
<p>To simplify the work, we will process the datasets one at a time. To align the first dataset, do the following:</p>
<pre><code>cd ~/workspace/module4
bismark --multicore 4 --bowtie2 $GENOME/genome/bismark_index \
    -1 $WGBS_DATA/WGBS.A34002.137160.chr19.1.fastq.gz -2 $WGBS_DATA/WGBS.A34002.137160.chr19.2.fastq.gz</code></pre>
<p><strong>Question 3</strong></p>
<p><em>What do all the options in the command mean? (Hint check the help by using <code>bismark --help</code>)</em></p>
<p><strong>Answer</strong></p>
<ul>
<li>The <code>--multicore 4</code> option is to do multithreaded processing to improve speed</li>
<li>The <code>--bowtie2</code> option is to use the mapping algorithm from bowtie2</li>
<li>The <code>$GENOME/genome/bismark_index</code> specifies the location of the index for the reference genome to use. This uses the <code>$GENOME</code> variable we defined previously</li>
<li>The -1 <code>$WGBS_DATA/WGBS.A34002.137160.chr19.1.fastq.gz specifies</code> the location of read 1. Idem for <code>-2</code> which specifies read 2. This uses the <code>$WGBS_DATA</code> variable we defined previously.</li>
</ul>
<p>For more details, please refer to the Bismark <a href="http://www.bioinformatics.babraham.ac.uk/projects/bismark/Bismark_User_Guide.pdf">user guide</a>.</p>
<p>This step will take a few minutes to run for this reduced dataset. A dataset spanning a full genome will take several hours.</p>
<p>For your own datasets, make sure you have enough computing walltime to run the alignment.</p>
<p><strong>While you wait for the results, ask any questions you have up to this point to the instructors.</strong></p>
<p>Check files</p>
<p>At the end of the alignment, you should have the following files saved into your workshop folder:</p>
<pre><code>WGBS.A34002.137160.chr19.1_bismark_bt2_pe.bam  WGBS.A34002.137160.chr19.1_bismark_bt2_PE_report.txt</code></pre>
<p>Let’s look at the report:</p>
<pre><code>less WGBS.A34002.137160.chr19.1_bismark_bt2_PE_report.txt</code></pre>
<p><strong>Question 4</strong></p>
<p><em>What was the mapping efficiency? What percent of C’s were methylated in CpG context?</em></p>
<p><strong>Answer</strong></p>
<p>According to the report:</p>
<pre><code>...
Mapping efficiency:     92.4% 
...
C methylated in CpG context:    57.4%
C methylated in CHG context:    0.6%
C methylated in CHH context:    0.5%
C methylated in unknown context (CN or CHN):    3.5%
...</code></pre>
<p>Close the report by pressing <code>q</code>.</p>
<p>Prepare files for loading in IGV</p>
<p>We need to sort the <code>bam</code> file and prepare an index so we will be able to load it in IGV. We will use the program <code>samtools</code> for this.</p>
<pre><code>samtools sort WGBS.A34002.137160.chr19.1_bismark_bt2_pe.bam -o WGBS.A34002.137160.chr19.1_bismark_bt2_pe_sorted.bam 
samtools index WGBS.A34002.137160.chr19.1_bismark_bt2_pe_sorted.bam</code></pre>
<p>Check Files</p>
<p>At the end, you should have the following files:</p>
<pre><code>WGBS.A34002.137160.chr19.1_bismark_bt2_pe_sorted.bam
WGBS.A34002.137160.chr19.1_bismark_bt2_pe.bam         
WGBS.A34002.137160.chr19.1_bismark_bt2_pe_sorted.bam.bai
WGBS.A34002.137160.chr19.1_bismark_bt2_PE_report.txt</code></pre>
</div>
<div id="repeat-alignment-for-all-datasets" class="section level4 hasAnchor">
<h4>2.3 Repeat Alignment for All Datasets<a href="module-4.html#repeat-alignment-for-all-datasets" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p><strong>Question 5</strong></p>
<p><em>How would you repeat the alignment with the other datasets?</em></p>
<p><strong>Answer</strong></p>
<p>This is the command to run bismark on the two other samples:</p>
<pre><code>cd ~/workspace/module4

bismark --multicore 4 --bowtie2 $GENOME/genome/bismark_index \
    -1 $WGBS_DATA/WGBS.A34002.137487.chr19.1.fastq.gz -2 $WGBS_DATA/WGBS.A34002.137487.chr19.2.fastq.gz
    
bismark --multicore 4 --bowtie2 $GENOME/genome/bismark_index \
    -1 $WGBS_DATA/WGBS.A34002.137488.chr19.1.fastq.gz -2 $WGBS_DATA/WGBS.A34002.137488.chr19.2.fastq.gz</code></pre>
<p>Remember, for the command to work, both <code>$GENOME</code> and <code>$WGBS_DATA</code> need to be defined.</p>
<p>This is the command to prepare the samples for IGV (sort and index):</p>
<pre><code>samtools sort  WGBS.A34002.137487.chr19.1_bismark_bt2_pe.bam -o WGBS.A34002.137487.chr19.1_bismark_bt2_pe_sorted.bam
samtools index WGBS.A34002.137487.chr19.1_bismark_bt2_pe_sorted.bam

samtools sort  WGBS.A34002.137488.chr19.1_bismark_bt2_pe.bam -o WGBS.A34002.137488.chr19.1_bismark_bt2_pe_sorted.bam
samtools index WGBS.A34002.137488.chr19.1_bismark_bt2_pe_sorted.bam</code></pre>
</div>
<div id="load-data-and-explore-using-igv" class="section level4 hasAnchor">
<h4>2.4 Load Data and Explore using IGV<a href="module-4.html#load-data-and-explore-using-igv" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>While you wait for the previous steps to finish executing, it is a good idea to begin exploring the alignments.</p>
<p>Copy Files to Your Local Computer to View in IGV (optional)</p>
<p>Retrieve the files called <code>WGBS.A34002.137160.chr19.1_bismark_bt2_pe_sorted.bam</code> and <code>WGBS.A34002.137160.chr19.1_bismark_bt2_pe_sorted.bam.bai</code> from the server using your internet browser and the public IP address of your AWS instance.</p>
<p>Launch IGV on your computer</p>
<p>If you haven’t installed it yet, please get it here <a href="http://software.broadinstitute.org/software/igv/download">IGV download</a>.</p>
<p>Make sure that the human genome is selected in the top left corner. It should read: <strong>Human (hg38)</strong>.</p>
<p>Load your sorted <code>bam</code> file in IGV using <code>File -&gt; Load from file</code>. <em>For this to work, you need to have the index file (<code>.bai</code>) in the same location as the <code>bam</code> file</em>. You can also load this directly from the internet using the URL of this file you got from your public IP page, using <code>File -&gt; Load from URL</code>.</p>
<p>Now, on IGV, go to the following location:</p>
<pre><code>chr19:43,375,889-45,912,052</code></pre>
<p>And zoom in until you see something.</p>
<p>For instance, try the following window:</p>
<pre><code>chr19:44,527,387-44,536,873</code></pre>
<p>You should see something like this:</p>
<p><img src="img/region2.png" /></p>
<p><strong>If it looks different, can you change the way the colors are displayed?</strong></p>
<p><strong>Which section of which chromosome is covered by this dataset?</strong></p>
<p><strong>Can you see any interesting patterns in the coverage?</strong></p>
</div>
<div id="generate-methylation-profiles" class="section level4 hasAnchor">
<h4>2.5 Generate Methylation Profiles<a href="module-4.html#generate-methylation-profiles" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>So far we have only mapped the reads using Bismark. We can generate methylation profiles using the following command:</p>
<pre><code>cd ~/workspace/module4

bismark_methylation_extractor --bedGraph WGBS.A34002.137160.chr19.1_bismark_bt2_pe.bam</code></pre>
<p><strong>Question 6</strong></p>
<p><em>How would you do the same for the other replicates?</em></p>
<p><strong>Answer</strong></p>
<p>These are the commands that you should use:</p>
<pre><code>cd ~/workspace/module4

bismark_methylation_extractor --bedGraph WGBS.A34002.137487.chr19.1_bismark_bt2_pe.bam
bismark_methylation_extractor --bedGraph WGBS.A34002.137488.chr19.1_bismark_bt2_pe.bam</code></pre>
<p>Make sure that <em>all</em> the files produced so far are displayed in your internet browser.</p>
<p><strong>While you wait for all the steps to finish, you can ask the instructors any questions you might have up until this point.</strong></p>
<p>Load all the files in IGV using <code>File -&gt; Load from file</code> or <code>File -&gt; Load from URL</code>.</p>
<p>At this point, if you load the region <code>chr19:44,527,387-44,536,873</code> you should see something like</p>
<p><img src="img/region2_full.png" /></p>
<p>This promoter looks to be hypomethylated.</p>
<p><em>Can you find a promoter that is hypermethylated?</em></p>
<p>How about <code>chr19:45,637,715-45,657,380</code>?</p>
<p><em>How would you look for a CpG island using this view of the data?</em></p>
<p>Keep exploring the files and see if you can find differences in the methylation profiles of the samples.</p>
</div>
</div>
<div id="differential-methylation-analysis-in-methylkit" class="section level3 hasAnchor">
<h3>3. Differential Methylation Analysis in MethylKit<a href="module-4.html#differential-methylation-analysis-in-methylkit" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The following section will use the Bioconductor package <code>methylKit</code> to do a differential methylation analysis. You can do it in your own computer (if you have installed <code>R</code> and <code>methylKit</code>) or in the AWS instance.</p>
<p>To install <code>methylKit</code> locally on your computer, make sure you have a recent version of <code>R</code> and follow the instructions in this page.</p>
<div id="load-r-and-methylkit" class="section level4 hasAnchor">
<h4>3.1 Load R and MethylKit<a href="module-4.html#load-r-and-methylkit" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>If you are working in AWS, you will need to load <code>R</code>. The image we provide already has the libraries we need.</p>
<p>To launch <code>R</code> simply type the following to your terminal:</p>
<pre><code>cd ~/workspace/module4
R</code></pre>
<p>If you did this properly, the following message will be displayed and your prompt will change from <code>ubuntu@ip-00-00-00-0:~/workspace/module4$</code> to <code>&gt;</code>:</p>
<pre><code>R version 4.0.5 (2021-03-31) -- &quot;Shake and Throw&quot;
Copyright (C) 2021 The R Foundation for Statistical Computing
Platform: x86_64-pc-linux-gnu (64-bit)
...</code></pre>
<p>Once you have successfully launched <code>R</code>, you can load <code>methylKit</code> with the following command:</p>
<pre><code>library(&quot;methylKit&quot;) </code></pre>
</div>
<div id="import-the-alignment-data-into-methylkit" class="section level4 hasAnchor">
<h4>3.2 Import the Alignment Data into MethylKit<a href="module-4.html#import-the-alignment-data-into-methylkit" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Process Bismark Alignments</p>
<p>To read the alignment data into <code>methylKit</code>, run the following command:</p>
<pre><code>methRaw.160 = processBismarkAln( location = &quot;WGBS.A34002.137160.chr19.1_bismark_bt2_pe_sorted.bam&quot;,
                         sample.id=&quot;A34002.137160&quot;, assembly=&quot;hg38&quot;, 
                         read.context=&quot;CpG&quot;, save.folder=&quot;methylkit&quot;)</code></pre>
<p>This command will import the data into a format that is readable by <code>methylKit.</code> At the same time, it will save two files under the <code>methylkit</code> directory with the information so that it is easy to load again at any time:</p>
<pre><code>methylkit/A34002.137160_CpG_conversionStats.txt
methylkit/A34002.137160_CpG.txt</code></pre>
<p>If everything goes well and you see the files, do the same for the other two samples:</p>
<pre><code>methRaw.487 = processBismarkAln( location = &quot;WGBS.A34002.137487.chr19.1_bismark_bt2_pe_sorted.bam&quot;,
                         sample.id=&quot;A34002.137487&quot;, assembly=&quot;hg38&quot;, 
                         read.context=&quot;CpG&quot;, save.folder=&quot;methylkit&quot;)
                         
methRaw.488 = processBismarkAln( location = &quot;WGBS.A34002.137488.chr19.1_bismark_bt2_pe_sorted.bam&quot;,
                         sample.id=&quot;A34002.137488&quot;, assembly=&quot;hg38&quot;, 
                         read.context=&quot;CpG&quot;, save.folder=&quot;methylkit&quot;)</code></pre>
<p>Create a MethylKit Object</p>
<p>Now that all the samples have been read with <code>methylKit</code>, you can create a file list to make it easier to load the full dataset as a <code>methylkit</code> object. For the purposes of this tutorial, we will consider that samples belong to two experimental groups: <code>A34002.137160</code> as the control group (treatment = 0) and <code>A34002.137487</code> &amp; <code>A34002.137488</code> as the treatment group (treatment = 1). We use the <code>methRead()</code> function to create our object, as shown below:</p>
<pre><code>file.list = list( file.path(&quot;methylkit&quot;, &quot;A34002.137160_CpG.txt&quot;),
                  file.path(&quot;methylkit&quot;, &quot;A34002.137487_CpG.txt&quot;),
                  file.path(&quot;methylkit&quot;, &quot;A34002.137488_CpG.txt&quot;) )

myobj = methRead(file.list,
           sample.id=list(&quot;A34002.137160&quot;,&quot;A34002.137487&quot;,&quot;A34002.137488&quot;),
           assembly=&quot;hg38&quot;,
           treatment=c(0,1,1),
           context=&quot;CpG&quot;,
           mincov = 10
           )</code></pre>
<p><strong>Question 7</strong></p>
<p><em>What do all the options in the <code>methRead()</code> command mean?</em></p>
<p><strong>Answer</strong></p>
<ul>
<li><p><code>file.list</code> object points to the location of the input data in a MethylKit format.</p></li>
<li><p><code>sample.id</code> points to a list with the appropriate sample name for each file.</p></li>
<li><p><code>assembly</code> specifies which build of the human reference genome is used.</p></li>
<li><p><code>treatment</code> specifies which sample belongs to each experimental group.</p></li>
<li><p><code>context</code> specifies the methylation context.</p></li>
<li><p><code>mincov</code> specifies the minimum coverage required to be included in the object.</p></li>
</ul>
<p>For more details, please refer to the MethylKit <a href="https://bioconductor.org/packages/release/bioc/manuals/methylKit/man/methylKit.pdf">user guide</a>.</p>
<p>If the files were loaded properly, you can check the object you just created by running the following command:</p>
<pre><code>myobj</code></pre>
<p>Which should output the following message followed by previews of the contents of the object:</p>
<pre><code>methylRawList object with 3 methylRaw objects
...</code></pre>
<p>You can also get basic statistics on your object by using the following command:</p>
<pre><code>getMethylationStats(myobj[[2]],plot=FALSE,both.strands=FALSE)</code></pre>
</div>
<div id="find-differentially-methylated-regions" class="section level4 hasAnchor">
<h4>3.3 Find Differentially Methylated Regions<a href="module-4.html#find-differentially-methylated-regions" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Merge Samples</p>
<p>Before doing any additional analysis, <code>methylKit</code> needs to determine which methylated bases have sufficient coverage in all samples so they can be compared. To do that, the samples should be merged with the <code>unite()</code> function. This function has a parameter <code>destrand=</code> that is turned off by default. We will set the <code>destrand</code> option to <code>TRUE</code> which will merge the coverage of both strands. When doing your own analyses, be aware that for some kinds of methylation analyses (such as CpH methylation) results are strand-specific, so this option should be used carefully.</p>
<pre><code>meth = unite(myobj, destrand=TRUE)</code></pre>
<p>Perform Differential Methylation Analysis</p>
<p>The standard function for Differential Methylation Analysis on <code>methylKit</code> is <code>calculateDiffMeth()</code>. It takes any <strong>merged</strong> methylkit object as input. Depending on the number of replicates, it uses either Fisher’s exact or logistic regression to calculate P-values. It also, automatically produces Q-values, which are a kind of adjusted P-value. To use it with the results we obtained before, run the following command:</p>
<pre><code>myDiff = calculateDiffMeth(meth)</code></pre>
<p>To check the output, just type <code>myDiff</code> and read the summary. If you want an example of the output, check the solution below.</p>
<p><strong>Example</strong></p>
<pre><code>This is what the output looks like:

methylDiff object with 2941 rows
--------------
    chr    start      end strand       pvalue     qvalue  meth.diff
1 chr19 42002896 42002896      + 3.271268e-01 0.69569299   8.951407
2 chr19 42002978 42002978      + 1.912989e-01 0.60732656 -21.666667
3 chr19 42007251 42007251      + 6.999764e-05 0.03228847 -55.681818
4 chr19 42007255 42007255      + 3.958578e-01 0.75196047 -11.835106
5 chr19 42007283 42007283      + 8.451850e-01 0.91347038  -2.457757
6 chr19 42007314 42007314      + 9.102723e-01 0.92865750  -1.604278
--------------
sample.ids: A34002.137160 A34002.137487 A34002.137488 
destranded TRUE 
assembly: hg38 
context: CpG 
treament: 0 1 1 
resolution: base </code></pre>
<p>To filter results by their statistical significance, <code>methylKit</code> provides the <code>getMethylDiff()</code> function which allows you to extract only the deferentially methylated CpG’s that meet a specific Q-value threshold. Additionally, it is also possible to specify whether to keep <code>hypo</code> or <code>hyper</code> methylated CpG’s only. Finally, the <code>bedgraph()</code> function allows you to save the the <code>methylDiff</code> object into a BedGraph file so you can open it with your genome browser of choice. Let’s create two BedGraph files with hypo and hyper methylated CpG’s with a Q-value below 0.05 based on the data above:</p>
<pre><code>myDiff.hyper = getMethylDiff(myDiff,qvalue=0.05,difference=10,type=&quot;hyper&quot;)
bedgraph(myDiff.hyper, file.name = &quot;hyper.CpG.bedGraph&quot;, col.name = &quot;qvalue&quot;)

myDiff.hypo = getMethylDiff(myDiff,qvalue=0.05,difference=10,type=&quot;hypo&quot;)
bedgraph(myDiff.hypo, file.name = &quot;hypo.CpG.bedGraph&quot;, col.name = &quot;qvalue&quot;)</code></pre>
<p>Two new files should appear now in your workshop folder:</p>
<pre><code>~/workspace/module4/hyper.CpG.bedGraph
~/workspace/module4/hypo.CpG.bedGraph</code></pre>
<p>Bin Results to Obtain Differentially Methylated Regions
By default, <code>methylKit</code> will compute results with an individual CpG resolution. To get Differentially Methylated Regions (<em>DMR</em>), you have to bin your results first, using a window size of your choice. The function to do this is <code>tileMethylCounts()</code>, which takes a regular methylkit object as input. In this case, we will create 1000bp bins using the following command:</p>
<pre><code>tiles = tileMethylCounts(myobj,win.size=1000,step.size=1000,cov.bases = 10)</code></pre>
<p>As with CpG level results, samples need to be merged before the analysis can continue:</p>
<pre><code>meth.tiles = unite(tiles, destrand=TRUE) </code></pre>
<p>Now, we will use the <code>calculateDiffMeth()</code> and <code>getMethylDiff()</code> functions to get the DMRs.</p>
<p><strong>Questions 8 &amp; 9</strong></p>
<p><em>Do you know how to do it, based on the information above?</em></p>
<p><em>Based on the number of differentially methylated CpGs you found above, do you anticipate many statistically significant DMRs in your analysis?</em></p>
<p><strong>Answer</strong></p>
<p>Use the following commands to perform a <code>DMR</code> analysis:</p>
<pre><code>myDiff.tiles = calculateDiffMeth(meth.tiles)

myDiff.tiles.hyper = getMethylDiff(myDiff.tiles,qvalue=0.1,difference=10,type=&quot;hyper&quot;)
bedgraph(myDiff.tiles.hyper, file.name = &quot;hyper.DMR.bedGraph&quot;, col.name = &quot;qvalue&quot;)

myDiff.tiles.hypo = getMethylDiff(myDiff.tiles,qvalue=0.1,difference=10,type=&quot;hypo&quot;)
bedgraph(myDiff.tiles.hypo, file.name = &quot;hypo.DMR.bedGraph&quot;, col.name = &quot;qvalue&quot;)</code></pre>
<p>Using the navigation pane, download the bedGraph files you just produced and try to open them with IGV.</p>
<p><em>Do the statistical results match what you had seen before when exploring the data?</em></p>
<p><em>What interesting genomic features are found close to the DMRs? What could this mean?</em></p>
<p>Congrats, you’re done!</p>
<p>You can quit R using the <code>quit()</code> or <code>q()</code> command. Remember to stop your AWS instance after this lab to avoid unnecessary costs.</p>
<p>Once you are finished make sure you download all the files you need and continue exploring on IGV.</p>
<div class="callout-green">
<span class="callout-icon"><i class="fas fa-circle-check" aria-hidden="true"></i></span>
<div class="callout-content">
<div class="callout-title-static">
<p>Lab Completed!</p>
</div>
<p>Congratulations! You have completed Lab 4!</p>
</div>
</div>

</div>
</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="module-3.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="module-5-downstream-analyses-integrative-tools.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
  "sharing": {
    "github": false,
    "facebook": false,
    "twitter": false,
    "linkedin": false,
    "weibo": false,
    "instapaper": false,
    "vk": false,
    "whatsapp": false,
    "all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
  },
  "fontsettings": {
    "theme": "white",
    "family": "sans",
    "size": 2
  },
  "edit": {
    "link": "https://github.com/bioinformaticsdotca/EPI_2021/blob/main/040-module-4.Rmd",
    "text": "Edit"
  },
  "history": {
    "link": null,
    "text": null
  },
  "view": {
    "link": null,
    "text": null
  },
  "download": ["_main.pdf", "_main.epub"],
  "search": {
    "engine": "fuse",
    "options": null
  },
  "toc": {
    "collapse": "subsection"
  }
});
});
</script>

</body>

</html>
