<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Module 2 | Epigenomic Analysis 2021</title>
  <meta name="description" content="Workshop website for the 2021 Epigenomic Analysis Canadian Bioinformatics Workshop" />
  <meta name="generator" content="bookdown 0.43 and GitBook 2.6.7" />

  <meta property="og:title" content="Module 2 | Epigenomic Analysis 2021" />
  <meta property="og:type" content="book" />
  <meta property="og:image" content="https://bioinformaticsdotca.github.io/EPI_2021/img/bioinformatics_logo.png" />
  <meta property="og:description" content="Workshop website for the 2021 Epigenomic Analysis Canadian Bioinformatics Workshop" />
  <meta name="github-repo" content="bioinformaticsdotca/EPI_2021" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Module 2 | Epigenomic Analysis 2021" />
  
  <meta name="twitter:description" content="Workshop website for the 2021 Epigenomic Analysis Canadian Bioinformatics Workshop" />
  <meta name="twitter:image" content="https://bioinformaticsdotca.github.io/EPI_2021/img/bioinformatics_logo.png" />

<meta name="author" content="Faculty: Guillaume Bourque, Martin Hirst, David Bujold, Jose Hector Galvez, Edmund Su, BF Francis Ouellette, Rachade Hmamouchi" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="img/favicon.ico" type="image/x-icon" />
<link rel="prev" href="module-1-introduction-to-chip-sequencing-and-analysis.html"/>
<link rel="next" href="module-3.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/6.5.1/css/all.min.css">
<script type="application/ld+json">
{
  "@context": "https://schema.org",
  "@type": "LearningResource",
  "name": "Epigenomics Analysis 2021",
  "description": "The complete set of tutorials, slides, and practical exercises for the Epigenomics Analysis 2021 Canadian Bioinformatics Workshop, hosted on GitHub Pages.",
  "url": "https://bioinformaticsdotca.github.io/EPI_2021",
  "learningResourceType": ["tutorial", "slides", "exercise"],
  "inLanguage": "en",
  "license": "https://creativecommons.org/licenses/by/4.0/",
  "provider": {
    "@type": "Organization",
    "name": "Bioinformatics.ca",
    "url": "https://bioinformatics.ca"
  },
  "keywords": "",
  "isPartOf": {
    "@type": "Course",
    "@id": "http://bioinformaticsdotca.github.io/EPI_2021"
  }
}
</script>



<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<a href="https://bioinformaticsdotca.github.io/">
 <img src="img/bioinformatics_logo.png" width="90%" alt="bioinformatics.ca logo" style="display:block; margin-left:auto; margin-right:auto; margin-top:15px;">
 </a>
<li><center><strong><a href="./"><br>Epigenomic Analysis<br><br></a></strong></center></li>
<a href="./">
 <img src="img/CBW_Epigenome-data_icon.png" width="70%" alt="Epigenomic Analysis logo" style="display:block; margin-left:auto; margin-right:auto; ">
 </a>
<br>

<li class="divider"></li>
<li class="part"><span><b>I Introduction</b></span></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Workshop Info</a>
<ul>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#pre-work"><i class="fa fa-check"></i>Pre-work</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#schedule"><i class="fa fa-check"></i>Schedule</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="meet-your-faculty.html"><a href="meet-your-faculty.html"><i class="fa fa-check"></i>Meet Your Faculty</a></li>
<li class="chapter" data-level="" data-path="compute-setup.html"><a href="compute-setup.html"><i class="fa fa-check"></i>Compute Setup</a></li>
<li class="part"><span><b>II Modules</b></span></li>
<li class="chapter" data-level="" data-path="module-1-introduction-to-chip-sequencing-and-analysis.html"><a href="module-1-introduction-to-chip-sequencing-and-analysis.html"><i class="fa fa-check"></i>Module 1: Introduction to ChIP Sequencing and Analysis</a>
<ul>
<li class="chapter" data-level="" data-path="module-1-introduction-to-chip-sequencing-and-analysis.html"><a href="module-1-introduction-to-chip-sequencing-and-analysis.html#lecture"><i class="fa fa-check"></i>Lecture</a></li>
<li class="chapter" data-level="" data-path="module-1-introduction-to-chip-sequencing-and-analysis.html"><a href="module-1-introduction-to-chip-sequencing-and-analysis.html#lab"><i class="fa fa-check"></i>Lab</a>
<ul>
<li class="chapter" data-level="" data-path="module-1-introduction-to-chip-sequencing-and-analysis.html"><a href="module-1-introduction-to-chip-sequencing-and-analysis.html#common-tools-of-the-trade"><i class="fa fa-check"></i>Common Tools of the Trade</a></li>
<li class="chapter" data-level="" data-path="module-1-introduction-to-chip-sequencing-and-analysis.html"><a href="module-1-introduction-to-chip-sequencing-and-analysis.html#module-1.-bwa-alignment-bam-post-processing"><i class="fa fa-check"></i>Module 1. BWA Alignment + BAM Post-processing</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="module-2.html"><a href="module-2.html"><i class="fa fa-check"></i>Module 2</a>
<ul>
<li class="chapter" data-level="" data-path="module-2.html"><a href="module-2.html#lecture-1"><i class="fa fa-check"></i>Lecture</a></li>
<li class="chapter" data-level="" data-path="module-2.html"><a href="module-2.html#lab-1"><i class="fa fa-check"></i>Lab</a>
<ul>
<li class="chapter" data-level="" data-path="module-2.html"><a href="module-2.html#module-2.-peak-calling"><i class="fa fa-check"></i>Module 2. Peak Calling</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="module-3.html"><a href="module-3.html"><i class="fa fa-check"></i>Module 3</a>
<ul>
<li class="chapter" data-level="" data-path="module-3.html"><a href="module-3.html#lecture-2"><i class="fa fa-check"></i>Lecture</a></li>
<li class="chapter" data-level="" data-path="module-3.html"><a href="module-3.html#lab-2"><i class="fa fa-check"></i>Lab</a>
<ul>
<li class="chapter" data-level="" data-path="module-3.html"><a href="module-3.html#visualization"><i class="fa fa-check"></i>3. Visualization</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="module-4.html"><a href="module-4.html"><i class="fa fa-check"></i>Module 4</a>
<ul>
<li class="chapter" data-level="" data-path="module-4.html"><a href="module-4.html#lecture-3"><i class="fa fa-check"></i>Lecture</a></li>
<li class="chapter" data-level="" data-path="module-4.html"><a href="module-4.html#lab-3"><i class="fa fa-check"></i>Lab</a>
<ul>
<li class="chapter" data-level="" data-path="module-4.html"><a href="module-4.html#introduction"><i class="fa fa-check"></i>1. Introduction</a></li>
<li class="chapter" data-level="" data-path="module-4.html"><a href="module-4.html#mapping-tutorial"><i class="fa fa-check"></i>2. Mapping Tutorial</a></li>
<li class="chapter" data-level="" data-path="module-4.html"><a href="module-4.html#differential-methylation-analysis-in-methylkit"><i class="fa fa-check"></i>3. Differential Methylation Analysis in MethylKit</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="module-5-downstream-analyses-integrative-tools.html"><a href="module-5-downstream-analyses-integrative-tools.html"><i class="fa fa-check"></i>Module 5: Downstream Analyses &amp; Integrative Tools</a>
<ul>
<li class="chapter" data-level="" data-path="module-5-downstream-analyses-integrative-tools.html"><a href="module-5-downstream-analyses-integrative-tools.html#lecture-4"><i class="fa fa-check"></i>Lecture</a></li>
<li class="chapter" data-level="" data-path="module-5-downstream-analyses-integrative-tools.html"><a href="module-5-downstream-analyses-integrative-tools.html#lab-4"><i class="fa fa-check"></i>Lab</a>
<ul>
<li class="chapter" data-level="" data-path="module-5-downstream-analyses-integrative-tools.html"><a href="module-5-downstream-analyses-integrative-tools.html#introduction-1"><i class="fa fa-check"></i>Introduction</a></li>
<li class="chapter" data-level="" data-path="module-5-downstream-analyses-integrative-tools.html"><a href="module-5-downstream-analyses-integrative-tools.html#lab-part-1"><i class="fa fa-check"></i>Lab Part 1</a></li>
<li class="chapter" data-level="" data-path="module-5-downstream-analyses-integrative-tools.html"><a href="module-5-downstream-analyses-integrative-tools.html#ihec-data-portal"><i class="fa fa-check"></i>IHEC Data Portal</a></li>
<li class="chapter" data-level="" data-path="module-5-downstream-analyses-integrative-tools.html"><a href="module-5-downstream-analyses-integrative-tools.html#predicting-motifs-with-homer"><i class="fa fa-check"></i>Predicting Motifs with HOMER</a></li>
<li class="chapter" data-level="" data-path="module-5-downstream-analyses-integrative-tools.html"><a href="module-5-downstream-analyses-integrative-tools.html#looking-for-go-terms-enrichment-with-great"><i class="fa fa-check"></i>Looking for GO Terms Enrichment with GREAT</a></li>
<li class="chapter" data-level="" data-path="module-5-downstream-analyses-integrative-tools.html"><a href="module-5-downstream-analyses-integrative-tools.html#going-back-to-homer-results"><i class="fa fa-check"></i>Going back to HOMER Results</a></li>
</ul></li>
</ul></li>
<li class="divider"></li>
<h1 id="sponsors">Sponsors</h1>
<center>
<a href="https://www.cihr-irsc.gc.ca"><img src="img/sponsors/cihr.png" width=90% style="padding-bottom: 20px" alt="Canadian Institutes of Health Research logo."></a>

<a href="https://genomecanada.ca/"><img src="img/sponsors/genomecanada.png" height="55" style="display:inline-block;"  style="padding-bottom: 20px" alt="Genome Canada logo.">
<a href="https://www.ontariogenomics.ca/"><img src="img/sponsors/ontariogenomics.png" height="55" style="display:inline-block;" style="padding-bottom: 20px;" alt="Ontario Genomics logo.">
<br><br>
<a href="https://oicr.on.ca/"><img src="img/sponsors/oicr.png" width=35% style="display:inline-block;" style="padding-bottom: 20px" alt="Ontario Institute for Cancer Research logo.">

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Epigenomic Analysis 2021</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="module-2" class="section level1 hasAnchor">
<h1>Module 2<a href="module-2.html#module-2" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div id="lecture-1" class="section level2 hasAnchor">
<h2>Lecture<a href="module-2.html#lecture-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<iframe src="https://drive.google.com/file/d/1OpvvIKXxbpgfROTdQ3aJyUggTa_dGjMU/preview" width="640" height="480" allow="autoplay">
</iframe>
</div>
<div id="lab-1" class="section level2 hasAnchor">
<h2>Lab<a href="module-2.html#lab-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="module-2.-peak-calling" class="section level3 hasAnchor">
<h3>Module 2. Peak Calling<a href="module-2.html#module-2.-peak-calling" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ol style="list-style-type: decimal">
<li>Dedup BAM file</li>
<li>Running a peak caller</li>
<li>Remove blacklist regions</li>
<li>Visualization</li>
</ol>
<div id="setup-1" class="section level4 hasAnchor">
<h4>Setup<a href="module-2.html#setup-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<pre><code>mkdir ~/workspace/peaks
samtools index -@4 ~/CourseData/EPI_data/Module1/MCF10A_resources/*.bam</code></pre>
<div id="dedup-bam-file" class="section level5 hasAnchor">
<h5>1. Dedup BAM File<a href="module-2.html#dedup-bam-file" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<p>Code:</p>
<pre><code>treatment_bam=~/CourseData/EPI_data/Module1/MCF10A_resources/MCF10A_H3K27ac.bam
treatment_dedup=~/workspace/peaks/MCF10A_H3K27ac.dedup.bam
samtools view -@4 ${treatment_bam} -bh -q10 -F1028 -o ${treatment_dedup}
input_bam=~/CourseData/EPI_data/Module1/MCF10A_resources/MCF10A_Input.bam
input_dedup=~/workspace/peaks/MCF10A_Input.dedup.bam
samtools view -@4 ${treatment_bam} -bh -q10 -F1028 -o ${treatment_dedup}</code></pre>
<p>Code Breakdown:</p>
<pre><code>samtools view -@4 ${treatment_bam} -bh -q10 -F1028 chr19# {&quot;-F1028&quot;: removes reads that unmapped and duplicated,
&quot;-@4&quot;: uses two threads,
&quot;-bh&quot;: outputs in binary,
&quot;q10&quot;: MAPQ must be &gt;=10</code></pre>
<p>Comments:</p>
<pre><code>This step will take a while, copies have been prepared in resources. Smallest file taking 3 minutes. Longest taking 10.</code></pre>
<ol start="2" style="list-style-type: decimal">
<li>Running a Peak Caller</li>
</ol>
<p>Code:</p>
<pre><code>mkdir -p ~/workspace/peaks
treatment_frag=~/CourseData/EPI_data/Module1/MCF10A_resources/MCF10A_H3K27ac.dedup.bam
input_frag=~/CourseData/EPI_data/Module1/MCF10A_resources/MCF10A_Input.dedup.bam
name=MCF10A_H3K27ac

macs3 callpeak -t ${treatment_frag} -c ${input_frag} -f BAMPE -g hs -n ${name} --keep-dup all --outdir ~/workspace/peaks/ --bdg 1&gt; ~/workspace/peaks/${name}.out.log 2&gt; ~/workspace/peaks/${name}.err.log</code></pre>
<p>Code Breakdown:</p>
<pre><code>macs3 callpeak \ #General purpose peak calling mode
-t ${treatment_frag} \ #Treatment file, can provide more than one to &quot;pool&quot;
-c ${input_frag} \ #Control File/Input
-f BAMPE \ #Instructs MACS2 on what kind of file to expect. Single/Paired-end bed/bam
-g hs \ #Sets appropriate genome size for background sampling hs=human, mm=humouse
-n ${name} \ #name
#-q 0.05 #FDR q value default
#-broad #&quot;broad mode&quot; for broad marks - stitches small peaks together
--outdir ~/workspace/peaks/ \ #where to output files otherwise stores in current working directory
--bdg \ #outputs pileup into bedgraphs
1&gt; ~/workspace/peaks/${name}.out.log \ #output log
2&gt;  ~/workspace/peaks/${name}.err.log #error log

cat  ~/workspace/peaks/${name}.err.lo</code></pre>
<p>Comments:</p>
<pre><code>List for common marks but is not exhaustive. Best practice would be to observe trends on genome browser and then decide.
Narrow Marks:
H3K27ac
H3K4me3

Broad Marks:
H3K27me3
H3K4me1
H3K36me3
H3K36me2
H3K9me3

Take a look at the err.log we see:
1. Summary of command (a good way of troubleshooting if the wrong paramters were provided
2. Progress of loading the two bams
3. Note the line &quot;mean fragment size is determined as 227.0 bp from treatment&quot;</code></pre>
<p>Output:
<em>See logs</em></p>
</div>
<div id="black-list-removalinspection" class="section level5 hasAnchor">
<h5>2. Black List Removal/Inspection<a href="module-2.html#black-list-removalinspection" class="anchor-section" aria-label="Anchor link to header"></a></h5>
<ul>
<li><em>2.1 Black list removal</em></li>
</ul>
<p>Code:</p>
<pre><code>blacklist=~/CourseData/EPI_data/Module1/QC_resources/hg38_blacklist.bed
sample=&quot;MCF10A_H3K27ac_peaks&quot;
bedtools intersect -v -a ~/workspace/peaks/${sample}.narrowPeak -b ${blacklist} &gt; ~/workspace/peaks/${sample}.blacklistRemoved.narrowPeak
bedtools intersect -u -a ~/workspace/peaks/${sample}.narrowPeak -b ${blacklist} &gt; ~/workspace/peaks/${sample}.blacklist.narrowPeak</code></pre>
<p>Output:
<em>No log. Bedtools completes Silently</em></p>
<ul>
<li><em>2.2 NarrowPeak inspection</em></li>
</ul>
<p>Code:</p>
<pre><code>sample=&quot;MCF10A_H3K27ac_peaks&quot;
wc -l ~/workspace/peaks/${sample}.blacklistRemoved.narrowPeak
head ~/workspace/peaks/${sample}.blacklistRemoved.narrowPeak -n5</code></pre>
<p>Output:</p>
<pre><code>32384 /home/ubuntu/workspace/peaks/MCF10A_H3K27ac_peaks.blacklistRemoved.narrowPeak
chr1    10003   10467   MCF10A_H3K27ac_peak_1   2160    .       47.2436 221.596 216.028 179
chr1    17283   17514   MCF10A_H3K27ac_peak_2   27      .       3.9983  5.1706  2.76136 82
chr1    180312  180997  MCF10A_H3K27ac_peak_3   803     .       28.8766 85.1306 80.3433 461
chr1    777953  778720  MCF10A_H3K27ac_peak_4   210     .       10.1154 24.4429 21.0334 503
chr1    778984  779223  MCF10A_H3K27ac_peak_5   44      .       4.87285 7.00538 4.45379 69</code></pre>
<p>Comments:</p>
<pre><code>chromosome name
peak start
peak stop
peak name
int(-10*log10pvalue)
N/A
Fold change at peak summit
-log10 P-value at Peak
-log10 Q-value at Peak
Summit position relative to peak</code></pre>
<ul>
<li><strong>2.3 Quality control via % of reads in key genomic areas</strong></li>
</ul>
<p>Code:</p>
<pre><code>sample=&quot;MCF10A_H3K27ac&quot;
query_bam=~/CourseData/EPI_data/Module1/MCF10A_resources/${sample}.bam
samtools view -@4 -q 10 -F 1028 $query_bam -c
samtools view -@4 -q 10 -F 1028 $query_bam -L ~/CourseData/EPI_data/Module1/QC_resources/encode_enhancers_liftover.bed -c
samtools view -@4 -q 10 -F 1028 $query_bam -L ~/workspace/peaks/${sample}_peaks.blacklistRemoved.narrowPeak -c</code></pre>
<p>Output:</p>
<pre><code>19632094
10752392
2028377</code></pre>
<p>Comments:</p>
<pre><code>samtools view #Viewing entirety or subset of BAM file
-q 10 \ #Parsing for reads with 5&gt;=MAPQ
-F 1028 \ # Remove reads that have the following flags:UNMAP,DUP #https://broadinstitute.github.io/picard/explain-flags.html
$query_bam \ # Bam of interest
-c # Count number of reads instead of displaying
-L # Region of interest. Bed File</code></pre>
<p>Enrichment at key regions:</p>
<table>
<colgroup>
<col width="25%" />
<col width="20%" />
<col width="18%" />
<col width="18%" />
<col width="17%" />
</colgroup>
<thead>
<tr class="header">
<th><strong>H3K4me1</strong></th>
<th><strong>H3K27me3</strong></th>
<th><strong>H3K27ac</strong></th>
<th><strong>H3K4me3</strong></th>
<th></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Reads Count</td>
<td></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr class="even">
<td>Total</td>
<td>71,545,746</td>
<td>55,438,916</td>
<td>19,632,094</td>
<td>26,381,511</td>
</tr>
<tr class="odd">
<td>Enhancer</td>
<td>48,466,245</td>
<td>25,742,977</td>
<td>10,752,392</td>
<td>12,302,265</td>
</tr>
<tr class="even">
<td>HOX</td>
<td>13,866</td>
<td>13,085</td>
<td>4,504</td>
<td>28,674</td>
</tr>
<tr class="odd">
<td>TSS</td>
<td>7,303,584</td>
<td>5,111,631</td>
<td>2,702,194</td>
<td>18,041,080</td>
</tr>
<tr class="even">
<td>In Peaks</td>
<td>27,539,876</td>
<td>25,072,097</td>
<td>2,028,377</td>
<td>20,920,281</td>
</tr>
<tr class="odd">
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr class="even">
<td></td>
<td></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr class="odd">
<td></td>
<td>H3K4me1</td>
<td>H3K27me3</td>
<td>H3K27ac</td>
<td>H3K4me3</td>
</tr>
<tr class="even">
<td>Percent of Total</td>
<td></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr class="odd">
<td>Total</td>
<td>100.00</td>
<td>100.00</td>
<td>100.00</td>
<td>100.00</td>
</tr>
<tr class="even">
<td>Enhancer</td>
<td>67.74</td>
<td>46.43</td>
<td>54.77</td>
<td>46.63</td>
</tr>
<tr class="odd">
<td>HOX</td>
<td>0.02</td>
<td>0.02</td>
<td>0.02</td>
<td>0.11</td>
</tr>
<tr class="even">
<td>TSS</td>
<td>10.21</td>
<td>9.22</td>
<td>13.76</td>
<td>68.39</td>
</tr>
<tr class="odd">
<td>In Peaks</td>
<td>38.49</td>
<td>45.22</td>
<td>10.33</td>
<td>79.30</td>
</tr>
</tbody>
</table>
<div class="callout-green">
<span class="callout-icon"><i class="fas fa-circle-check" aria-hidden="true"></i></span>
<div class="callout-content">
<div class="callout-title-static">
<p>Lab Completed!</p>
</div>
<p>Congratulations! You have completed Lab 2!</p>
</div>
</div>

</div>
</div>
</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="module-1-introduction-to-chip-sequencing-and-analysis.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="module-3.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
  "sharing": {
    "github": false,
    "facebook": false,
    "twitter": false,
    "linkedin": false,
    "weibo": false,
    "instapaper": false,
    "vk": false,
    "whatsapp": false,
    "all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
  },
  "fontsettings": {
    "theme": "white",
    "family": "sans",
    "size": 2
  },
  "edit": {
    "link": "https://github.com/bioinformaticsdotca/EPI_2021/blob/main/020-module-2.Rmd",
    "text": "Edit"
  },
  "history": {
    "link": null,
    "text": null
  },
  "view": {
    "link": null,
    "text": null
  },
  "download": ["_main.pdf", "_main.epub"],
  "search": {
    "engine": "fuse",
    "options": null
  },
  "toc": {
    "collapse": "subsection"
  }
});
});
</script>

</body>

</html>
